### PR TITLE
Fix ca2208 errors

### PR DESCRIPTION
**Pull request by Jimmys20**
This PR fixes all [CA2208](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2208) violations and enables a rule so that they won't get violated again in the future.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/iot/pull/1844)

### CODE REVIEW

**Comment by pgrawehr** [comment]
Thanks for the submission. I can't seem to find where you enabled that rule, though. Can you make sure the rule is enabled so that any future builds would fail if it gets violated again?

File: src/devices/SoftwareSpi/SoftwareSpi.cs [hunk 1]
```diff
@@ -145,7 +145,7 @@ public override void TransferFullDuplex(ReadOnlySpan<byte> dataToWrite, Span<byt
         {
             if (dataToWrite.Length != dataToRead.Length)
             {
-                throw new ArgumentException(nameof(dataToWrite));
+                throw new ArgumentException(null, nameof(dataToWrite));
```
Comment (pgrawehr) [inline src/devices/SoftwareSpi/SoftwareSpi.cs:148]: I'm not sure passing null here is a good idea. Wouldn't this possibly lead to a NullReferenceException when handling the error?

Comment (Jimmys20) [reply src/devices/SoftwareSpi/SoftwareSpi.cs:148]: This is what visual studio code fix does by default. I could replace null with actual message describing what caused the exception. 

ca2208-codefix_null_msg

Comment (pgrawehr) [reply src/devices/SoftwareSpi/SoftwareSpi.cs:148]: That would be the ideal case, for sure.

Comment (Jimmys20) [reply src/devices/SoftwareSpi/SoftwareSpi.cs]: Replaced null parameters with error messages.

**Comment by Jimmys20** [comment]
I added the rule in editorconfig with severity level error.

File: .editorconfig [hunk 2]
```diff
@@ -135,6 +135,9 @@ csharp_space_between_method_declaration_parameter_list_parentheses = false
 csharp_space_between_parentheses = false
 csharp_space_between_square_brackets = false
 
+# CA2208: Instantiate argument exceptions correctly
+dotnet_diagnostic.CA2208.severity = error
```
Comment (pgrawehr) [inline .editorconfig:139]: Sorry for not being precise enough. This only affects VS, but not the build on the command line. Add the warning to the file `src/StyleCop.Analyzers.ruleset` instead. Add a line with `<Rule Id="CA2208" Action="Error" />`.

Comment (Jimmys20) [reply .editorconfig]: No worries, should be ok now.

**Review by pgrawehr — APPROVED** [review]

**Comment by krwq** [comment]
@Jimmys20 there seem to be some merge conflict here but otherwise it can be merged

**Pull request merged**